pkg/landscaper/execution/reconcile.go

```diff
@@ -49,26 +53,28 @@ func (o *Operation) Reconcile(ctx context.Context) lserrors.LsError {
 
 	allSucceeded := true
 	for _, item := range executionItems {
-		dlogger := logger.WithValues("deployitem", item.Info.Name)
-		if deployItemUpToDateAndNotForceReconcile(o.exec, item.Info.Name, item.DeployItem, o.forceReconcile, dlogger) {
+		diLogger := logger.WithValues("deployitem", item.Info.Name)
+		diCtx := logging.NewContext(ctx, diLogger)
+
+		if deployItemUpToDateAndNotForceReconcile(diCtx, o.exec, item.Info.Name, item.DeployItem, o.forceReconcile) {
 			if failedBecausePickupTimeout(item.DeployItem) {
 				// we need to check if the deploy item is failed due to that it is not picked up by a deployer.
 				// This is an error scenario that needs special handling as other detections in this code will not work.
 				// Mostly due to outdated observed generation
-				setPhaseFailedBecausePickupTimeout(o.exec, cond, item.Info.Name, item.DeployItem.Name, dlogger)
+				setPhaseFailedBecausePickupTimeout(diCtx, o.exec, cond, item.Info.Name, item.DeployItem.Name)
 				return nil
 			} else if failedAndStatusUpToDate(item.DeployItem) {
 				// the deployitem up-to-date
 				// deployitem is failed => set execution to failed
-				setPhaseFailedBecauseFailedDeployItem(o.exec, cond, item.Info.Name, item.DeployItem.Name, dlogger)
+				setPhaseFailedBecauseFailedDeployItem(diCtx, o.exec, cond, item.Info.Name, item.DeployItem.Name)
 				return nil
 			} else if notCompletedPhaseOrNotStatusUpToDate(item.DeployItem) {
 				// deployitem is running - either in a non-final phase, or its observedGeneration doesn't match its generation
-				setPhaseProgressingOfRunningDeployItem(o.exec, item.Info.Name, o.EventRecorder(), dlogger)
+				setPhaseProgressingOfRunningDeployItem(diCtx, o.exec, item.Info.Name, o.EventRecorder())
 				allSucceeded = false
 			} else {
 				// the deployitem is: up-to-date, in a final state, not failed => deployItem.spec.phase == succeeded => nothing to do with the deployitem
-				dlogger.Debug("deployitem not triggered because up-to-date", "deployItemPhase", string(item.DeployItem.Status.Phase))
+				diLogger.Debug("deployitem not triggered because up-to-date", "deployItemPhase", string(item.DeployItem.Status.Phase))
```

Comment:
do we need a constant for deployItemPhase?

Reply:
There are constants for the `Status.Phase` field for each resource (Installation, Execution, DeployItem). I didn't add one for the new deployitemPhase, because I didn't know how I should name it. Also, I personally would prefer to get rid of that field again, because I find it really confusing to have two phases with nearly the same purpose and no obvious distinction in the DeployItem status.

Reply:
Hi @Diaphteiros, Robert and I already planned to remove the additional phase field but perhaps in another PR.

Reply:
I am using now the `KeyDeployItemPhase` constant.